backend/app/app/static/js/calendar.js

```diff
@@ -72,8 +72,10 @@ function enroll(eventClickInfo) {
                     goto_profile();
                 } else {
                     switch (data.error.code) {
-                        // Have already been enrolled
-                        case 1:
+                        // Not a student account
+                        case 3:
+                        // Deadline passed
+                        case 4:
                             goto_profile();
                             break;
```

Comment:
If fall down supposed here?
Hitting `data.error.code === 3` will execute `goto_profile`  due to lack of `break` statement

Reply:
Yes, it is supposed to work like that, for both error codes 3 and 4